### PR TITLE
General maintenance: readme badges, autoformatting, flake8 test, .gitignore, changelog

### PR DESCRIPTION
**Pull request by consideRatio**
I did a general maintenance pass at the repo.

- [x] Readme in markdown with the common badges (Transitioned to markdown)
- [x] Autoformatting with black for python and prettier for markdown and yaml
- [x] Flake8 for Python code validation
- [x] pre-commit for automating black / prettier / flake8
- [x] Boilerplate Python.gitignore
- [x] Changelog for the next release

I figure it make sense to release `1.1` at this point as we added some new functionality to support internal_ssl in #11. Okay to cut a `1.1` release?

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    rev: v2.2.1
+    hooks:
+      - id: prettier
+        exclude: COPYING.md
+
+  # Lint: Python code
+  - repo: https://gitlab.com/pycqa/flake8
+    rev: "3.8.4"
+    hooks:
+      - id: flake8
```
Comment (manics) [inline .pre-commit-config.yaml:30]: I like the combination of black/prettier/flake8.
@minrk suggested some others in https://github.com/jupyterhub/kubespawner/pull/473/files#diff-63a9c44a44acf85fea213a857769990937107cf072831e1a26808cfde9d096b9 (copied from jupyterhub), do you think it's worth adding those too and trying to standardise this across all Python repos, or are we better off leaving it to each repo?

**Review by manics — COMMENTED** [review]
Couple of minor comments but looks fine to me.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (consideRatio) [reply .pre-commit-config.yaml:30]: > do you think it's worth adding those too and trying to standardise this across all Python repos, or are we better off leaving it to each repo?

Can we delegate this to a team-compass issue? I'd like to stay clear of more work to consider those as part of this PR, but I know that prettier have done a great job with json / yaml for example, and even within markdown files that has code blocks of json / yaml - so various of those pre-commit hooks become irrelevant I think.

Very positive to prettier as an autoformatter with the piloting epxerience in z2jh btw!

File: .github/workflows/test.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  #   steps:
+  #     - uses: actions/checkout@v2
+  #     - uses: actions/setup-python@v2
+  #       with:
+  #         python-version: "${{ matrix.python-version}}"
+
+  #     - name: Install Python dependencies
+  #       run: |
+  #         pip install pytest
+  #         pip install .
```
[the remark this answers is not in the document]
Comment (consideRatio) [reply .github/workflows/test.yaml]: Addressed in a3cac6a!

File: .pre-commit-config.yaml (hunk 1 above)
Comment (consideRatio) [reply .pre-commit-config.yaml:30]: I went ahead and added some misc content anyhow in b508b75

check-yaml / check-json is covered by prettier, and flake8 is deprecated and we are using the new dedicated pre-commit hook in the flake8 official repo.

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    # ref: https://github.com/pre-commit/pre-commit-hooks#hooks-available
+    hooks:
+      # Autoformat: Makes sure files end in a newline and only a newline.
+      - id: end-of-file-fixer
+
+      # Autoformat: Sorts entries in requirements.txt.
+      - id: requirements-txt-fixer
+
+      # Prevent giant (500kB) files from being committed.
+      - id: check-added-large-files
```
Comment (minrk) [inline .pre-commit-config.yaml:44]: What does this one solve? There is a cost to every hook, and I don't see this one coming up and doing the right thing.

File: .pre-commit-config.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    rev: 20.8b1
+    hooks:
+      - id: black
+        args: [--target-version=py36]
+
+  # Autoformat: markdown, yaml
+  - repo: https://github.com/pre-commit/mirrors-prettier
+    rev: v2.2.1
+    hooks:
+      - id: prettier
```
Comment (minrk) [inline .pre-commit-config.yaml:23]: WDYT about adding nodejs as a dependency to simply *contribute* to pure-Python projects? I love prettier, but I have worried about this. Should we wait for feedback from contributors? The files affected in this repo are pretty minimal.

**Review by minrk — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 4 above)
Comment (manics) [reply .pre-commit-config.yaml:23]: https://pre-commit.com/#node says 
>  node hooks work without any system-level dependencies. It has been tested on linux and macOS and may work under cygwin.

Though I haven't checked how this behaves in practice

File: .pre-commit-config.yaml (hunk 3 above)
Comment (consideRatio) [inline .pre-commit-config.yaml:44]:
```suggestion
```

Comment (consideRatio) [reply .pre-commit-config.yaml]: Not much :p deleted!

File: .pre-commit-config.yaml (hunk 4 above)
Comment (consideRatio) [reply .pre-commit-config.yaml:23]: If we required adding nodejs as a dependency to contribute to a python project, that would be a serious issue in my mind as well.

I have not explicitly tried without having node, but I'm happy to read what @manics quoted.

Comment (minrk) [reply .pre-commit-config.yaml:23]: Neat! I just tested it in an empty docker environment, and it uses https://github.com/ekalinin/nodeenv to get node itself. pre-commit set up the prettier environment (including getting node itself!)  quicker than the env for black, so I don't think anyone should notice the runtime. This is wonderful news!

**Comment by minrk** [comment]
Great!

**Comment by consideRatio** [comment]
Thanks @minrk !!

**Comment by plockaby** [comment]
Are we able to cut a release now? I'd like to resolve #18

**Comment by consideRatio** [comment]
Thanks for the nudge @plockaby!

**Comment by plockaby** [comment]
Looks great, thanks so much!

**Pull request merged**